# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1363030606 14400
#      Mon Mar 11 15:36:46 2013 -0400
# Branch classdef
# Node ID 0dc6731fc0da942006e58d24486be84ec562221d
# Parent  ea78011f66bf711efc6f9f5de85c0fdfd05286e9
avoid eat_continuation in classdef branch

* lex.ll (octave_lexer::handle_superclass_identifier,
octave_lexer::handle_meta_identifier): Don't strip trailing whitespace
from yytext since the pattern no longer allows it.  Don't call
eat_continuation.

diff --git a/libinterp/parse-tree/lex.ll b/libinterp/parse-tree/lex.ll
--- a/libinterp/parse-tree/lex.ll
+++ b/libinterp/parse-tree/lex.ll
@@ -2457,24 +2457,17 @@ octave_lexer::looks_like_command_arg (vo
 
   return (space_before && ! space_after
           && previous_token_may_be_command ());
 }
 
 int
 octave_lexer::handle_superclass_identifier (void)
 {
-  char *yytxt = flex_yytext ();
-  int c = yytxt[flex_yyleng()-1];
-
-  std::string meth = strip_trailing_whitespace (yytxt);
-
-  int cont_is_spc = eat_continuation ();
-
-  int spc_gobbled = (cont_is_spc || c == ' ' || c == '\t');
+  std::string meth = flex_yytext ();
 
   size_t pos = meth.find ("@");
   std::string cls = meth.substr (pos + 1);
   meth = meth.substr (0, pos);
 
   std::string pkg;
   pos = cls.find (".");
   if (pos != std::string::npos)
@@ -2497,24 +2490,17 @@ octave_lexer::handle_superclass_identifi
   current_input_column += flex_yyleng ();
 
   return SUPERCLASSREF;
 }
 
 int
 octave_lexer::handle_meta_identifier (void)
 {
-  char *yytxt = flex_yytext ();
-  int c = yytxt[flex_yyleng()-1];
-
-  std::string cls = strip_trailing_whitespace (yytxt).substr (1);
-
-  int cont_is_spc = eat_continuation ();
-
-  int spc_gobbled = (cont_is_spc || c == ' ' || c == '\t');
+  std::string cls = std::string(flex_yytext ()).substr (1);
 
   std::string pkg;
   size_t pos = cls.find (".");
   if (pos != std::string::npos)
     {
       pkg = cls.substr (0, pos);
       cls = cls.substr (pos + 1);
     }
